wordpress-mobile/WordPress-Android
# Self-hosted user UI

In this PR, I've combined #21264 (self-hosted user list) and #21270 (self-hosted user detail). This did make for a larger PR, but it should be easier to review than having separate PRs, plus a lot of the code is Jetpack Compose which can be quite verbose.

To test:

- Enable `self_hosted_users` in debug settings
- Switch to a self-hosted site
- Note that a new "Users" item appears
- Tap on that and verify the list screen with dummy data looks fine
- Tap a user to view the detail screen and verify that looks fine
- If the user has an avatar, tap it and verify the large avatar view looks fine
- Switch to a non-self-hosted site and verify the "Users" item doesn't appear
- View the various Compose previews of the `SelfHostedUsersScreen` and verify they look fine (note that remote avatar images won't load in the preview)

Caveats:

- This is a UI-only PR - no network requests are made, instead only dummy data is used
- The UI is somewhat different than in [this iOS PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/23572) but it's closer to the appearance of the existing People feature on Android

https://github.com/user-attachments/assets/14686212-5976-44cf-abcb-a84238973fbf
